### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by MartinDuPlooy1** [OCSInventory-NG/OCSInventory-ocsreports#1053]
**OCS Inventory version**
Version : 2.8

**Describe the bug**
Not all agents work after Upgrade to 2.8
Apache Log

192.168.x.x - - [20/Oct/2020:10:03:57 +1300] "POST /ocsinventory HTTP/1.1" 400 226 "-" "OCS-NG_WINDOWS_AGENT_v2.6.0.0"

172.16.x.x - - [20/Oct/2020:10:04:03 +1300] "POST /ocsinventory HTTP/1.1" 400 226 "-" "OCS-NG_WINDOWS_AGENT_v2.8.0.0"

172.16.x.x - - [20/Oct/2020:10:04:18 +1300] "POST /ocsinventory HTTP/1.1" 500 536 "-" "OCS-NG_WINDOWS_AGENT_v2.3.1.1"

172.16.x.x - - [20/Oct/2020:10:04:18 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

172.16.x.x - - [20/Oct/2020:10:04:24 +1300] "POST /ocsinventory HTTP/1.1" 200 - "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

172.16.x.x - - [20/Oct/2020:10:04:26 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

172.16.x.x - - [20/Oct/2020:10:04:29 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.3.1.1"

172.16.x.x - - [20/Oct/2020:10:04:31 +1300] "POST /ocsinventory HTTP/1.1" 500 536 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

192.168.x.x - - [20/Oct/2020:10:04:43 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

200 232192.168.x.x - - [20/Oct/2020:10:04:54 +1300] "POST /ocsinventory HTTP/1.1" 500 536 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

192.168.x.x - - [20/Oct/2020:10:05:10 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

172.16.x.x - - [20/Oct/2020:10:05:16 +1300] "POST /ocsinventory HTTP/1.1" 400 226 "-" "OCS-NG_WINDOWS_AGENT_v2.8.0.0"

172.16.x.x - - [20/Oct/2020:10:05:19 +1300] "POST /ocsinventory HTTP/1.1" 500 536 "-" "OCS-NG_WINDOWS_AGENT_v2.3.1.1"

192.168.x.x - - [20/Oct/2020:10:05:21 +1300] "POST /ocsinventory HTTP/1.1" 500 536 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

172.16.x.x - - [20/Oct/2020:10:05:39 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.3.1.1"

172.16.x.x - - [20/Oct/2020:10:06:01 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

172.16.x.x - - [20/Oct/2020:10:06:07 +1300] "POST /ocsinventory HTTP/1.1" 500 536 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

172.16.x.x - - [20/Oct/2020:10:06:12 +1300] "POST /ocsinventory HTTP/1.1" 200 - "-" "OCS-NG_WINDOWS_AGENT_v2.3.1.1"

192.168.x.x - - [20/Oct/2020:10:06:32 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

192.168.x.x - - [20/Oct/2020:10:06:58 +1300] "POST /ocsinventory HTTP/1.1" 500 536 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

192.168.x.x - - [20/Oct/2020:10:06:59 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

192.168.x.x - - [20/Oct/2020:10:07:43 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

192.168.x.x - - [20/Oct/2020:10:07:46 +1300] "POST /ocsinventory HTTP/1.1" 200 232 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

192.168.x.x - - [20/Oct/2020:10:07:55 +1300] "POST /ocsinventory HTTP/1.1" 500 536 "-" "OCS-NG_WINDOWS_AGENT_v2.4.0.0"

OCS Activity Log

Tue Oct 20 10:26:46 2020;2856;100;P170805-2020-10-19-08-52-58;172.16.x.x;OCS-NG_WINDOWS_AGENT_v2.3.1.1;prolog;accepted

Tue Oct 20 10:26:46 2020;2856;311;P170805-2020-10-19-08-52-58;172.16.x.x;OCS-NG_WINDOWS_AGENT_v2.3.1.1;session;started

Tue Oct 20 10:26:52 2020;2857;319;P170805-2020-10-19-08-52-58;172.16.x.x;OCS-NG_WINDOWS_AGENT_v2.3.1.1;session;found

Tue Oct 20 10:26:52 2020;2857;104;P170805-2020-10-19-08-52-58;172.16.x.x;OCS-NG_WINDOWS_AGENT_v2.3.1.1;inventory;incoming

Tue Oct 20 10:26:52 2020;2857;110;P170805-2020-10-19-08-52-58;172.16.x.x;OCS-NG_WINDOWS_AGENT_v2.3.1.1;inventory_cache;checking

Tue Oct 20 10:26:52 2020;2857;108;P170805-2020-10-19-08-52-58;172.16.x.x;OCS-NG_WINDOWS_AGENT_v2.3.1.1;inventory_cache;cache(softwares.NAME)

Tue Oct 20 10:26:52 2020;2857;522;P170805-2020-10-19-08-52-58;172.16.x.x;OCS-NG_WINDOWS_AGENT_v2.3.1.1;inventory_cache;fault:softwares.NAME

Tue Oct 20 10:26:52 2020;2857;113;P170805-2020-10-19-08-52-58;172.16.x.x;OCS-NG_WINDOWS_AGENT_v2.3.1.1;inventory;u:softwares

Tue Oct 20 10:26:52 2020;2857;515;P170805-2020-10-19-08-52-58;172.16.x.x;OCS-NG_WINDOWS_AGENT_v2.3.1.1;end;error

I see many of the end;error
My OCS GUI also now create empty entries;

MySQL Log, getting a lot of Rollbacks
  105 Query     SELECT OSNAME FROM hardware_osname_cache WHERE OSNAME='Microsoft Windows 10 Pro'
                  105 Query     SHOW COLUMNS FROM accountinfo
                  105 Query     SELECT ID,HARDWARE_ID,LANGUAGE,NAME,PUBLISHER,COMMENTS,INSTALLDATE,CATEGORY,GUID,FILESIZE,VERSION,FOLDER,FILENAME,BITSWIDTH,SOURCE FROM softwares
      WHERE HARDWARE_ID='11666' ORDER BY NAME
                  105 Query     DELETE FROM locks WHERE HARDWARE_ID='11666' AND ID='2858'
                  105 Query     rollback
                  105 Query     rollback

I also notices that some queries search from Tables Softwares and not Software. My OCS database doesn't have a softwares Table? List of all my Tables in OCS Database

accesslog
accountinfo
accountinfo_config
archive
assets_categories
batteries
bios
blacklist_macaddresses
blacklist_serials
blacklist_subnet
config
conntrack
controllers
cpus
cve_search
cve_search_correspondance
cve_search_history
deleted_equiv
deploy
devices
devicetype
dico_ignored
dico_soft
download_affect_rules
download_available
download_enable
download_history
download_servers
downloadwk_conf_values
downloadwk_fields
downloadwk_history
downloadwk_pack
downloadwk_statut_request
downloadwk_tab_values
drives
engine_mutex
engine_persistent
extensions
files
groups
groups_cache
hardware
hardware_osname_cache
inputs
itmgmt_comments
javainfo
journallog
languages
locks
memories
modems
monitors
netmap
network_devices
networks
networkshare
notification
notification_config
officepack
operators
ports
printers
prolog_conntrack
regconfig
registry
registry_name_cache
registry_regvalue_cache
repository
saas
saas_exp
save_query
schedule_WOL
securitycenter
service
sim
slots
snmp
snmp_accountinfo
snmp_blades
snmp_cards
snmp_cartridges
snmp_communities
snmp_computers
snmp_configs
snmp_cpus
snmp_drives
snmp_fans
snmp_firewalls
snmp_inputs
snmp_labels
snmp_laststate
snmp_links
snmp_loadbalancers
snmp_localprinters
snmp_memories
snmp_mibs
snmp_modems
snmp_networks
snmp_ports
snmp_powersupplies
snmp_printers
snmp_softwares
snmp_sounds
snmp_storages
snmp_switchinfos
snmp_switchs
snmp_trays
snmp_types
snmp_videos
snmp_virtualmachines
software
software_categories
software_category_exp
software_name
software_publisher
software_version
softwares_name_cache
sounds
ssl_store
storages
subnet
tags
temp_files
uptime
usbdevices
videos
virtualmachines
winupdatestate
winusers

I am also not able to find the HARDWARE_ID=11666?
Why am I getting these errors? DB access is fine by the way.
Please advise.

**To Reproduce**
See Logs. 

**Expected behavior**
All clients connect and update OCS.

**Screenshots**
![OCS_Empty](https://user-images.githubusercontent.com/73196054/96635094-ed683e80-1377-11eb-83ee-a86a03d5c94b.png)

**Log file (optional)**
Apache / PHP / System log if needed

[OCS.zip](https://github.com/OCSInventory-NG/OCSInventory-ocsreports/files/5411123/OCS.zip)

**Comment by dit-aix** [comment]
hi, i got multiples bug with 2.8 here too
@MartinDuPlooy1 The softwares tables doesn't exist anymore, it has been replaced by a software table and somes software_xxx tables (see files 7030.sql to 7037.sql in ocsreports/files/update from OCSNG_UNIX_SERVER_2.8 to understand)

On my server, i updated from 2.6 to 2.7 then immediatly to 2.8. During the update process something gone wrong because my new software table was empty .... I had to restore the old softwares table from a backup and manually run updates.sql into the mysql client command line. During the initial update (install.php) the installer take a long time to update the database (maybe a timeout somewhere during this stage ?) i had more than 2.5M rows in softwares table.

so here the 2.6 client still works (about 4000 inventory during last 24h) except for softwares because the sql query try to insert into "softwares" that doesn't exist anymore. I think i need to update the client to 2.8 to insert into software table.

But 2.8 client doesn't work at all, see https://github.com/OCSInventory-NG/WindowsAgent/issues/199 for more information.

I tried to generate a local file with the client (2.6 and 2.8) and import it manualy through the ocsreports GUI and i got : Erreur HTTP/1.1 403 Forbidden 

So for me, 2.8 client and/or 2.8 server have a bug in inventory upload.
